ws-nextjs-app/pages/[service]/send/[id]/FormContext/index.tsx

```diff
@@ -78,17 +80,25 @@ export const FormContextProvider = ({
   const [submitted, setSubmitted] = useState(false);
   const [progress, setProgress] = useState('0');
   const [submissionError, setSubmissionError] = useState<SubmissionError>(null);
+  const [attemptCount, setAttemptCount] = useState(0);
 
   const handleChange = (name: OnChangeInputName, value: OnChangeInputValue) => {
     setFormState(prevState => {
-      const updatedState = { [name]: { ...prevState[name], value } };
+      const currState = { ...prevState[name], value };
+      const validateFunction = validateFunctions[currState.htmlType];
+      const validatedData = validateFunction
+        ? validateFunction(currState)
+        : currState;
+
+      const updatedState = { [name]: { ...validatedData } };
       return { ...prevState, ...updatedState };
     });
   };
 
   const handleSubmit = async (event: FormEvent) => {
     event.preventDefault();
     setSubmitted(true);
+    setAttemptCount(currCount => currCount + 1);
```

Comment:
Does this need to be a number? Not sure we really care too much about how many times the user has attempted to submit, just whether they have or not so would a boolean work here?